### DISCUSSION TITLE
v1.21 feedback thread

### DISCUSSION THREAD

**Discussion by Lailloken** [Lailloken/Exile-UI#33] [Ideas & Feedback]
Feedback thread for v1.21, highlighted changes:
- custom resolutions
- tighter scanner settings
- UI changes: prio-list & ready-panel
- pausing prio-recipes

**Comment by CarryOnMyWayward** [comment]
It's me again, dude with 1360x768 :D Tried forced resolution, works good so far, Only 1 scan per mod, didn't break Trade Companion or Awakened Trade Macro (but 48 pixels less for price list, yeah). Nice job!

**Comment by FeralRage7** [comment]
Thank you for implementing the pre-game start for the script. I personally appreciate it!

**Reply by FeralRage7** [reply]
@Lailloken also, what is supposed to happen when I click the recipe on the right hand pop out and it turns purple?

**Reply by Lailloken** [reply]
This 'pauses' tracking the recipe. You will not get suggestions for it, which will prevent it from taking sub-components away from other prio-recipes. You can use it if a prio-recipe pulls ahead of others or if you feel like you have assembled enough. This is different from removing it from the prio-list in that its components don't go into the burn pool but stay prioritized.

**Comment by OmikronGamma** [comment]
Not fan of new version. 
Cant see list of missing ingredients.
Clicking on archenemsis name on prio list (the top one) doesnt show its recipe.

**Reply by FeralRage7** [reply]
@OmikronGamma I think if you click and hold on the prio list, you will see ingredients. Can you try that?

**Reply by OmikronGamma** [reply]
Nope, it shows ingredients to burn list.

**Reply by Lailloken** [reply]
> Cant see list of missing ingredients.

Do you mean the icons that used to be at the top? I removed them because they seemed redundant with the new map suggestions panel. I assumed just following that and running optimal maps was better than checking the icons individually. The ready panel now tells you how close you are to completion of the prio-list (X bases missing). If you only want to loot missing bases, you can shift-click the MISSING-label to open a movable list that acts as a cheat sheet.

> Clicking on archenemsis name on prio list (the top one) doesnt show its recipe.

Hold-clicking shows the tree, normal clicking highlights the assembled version if one is available in the inventory. The only way to ~show~ (EDIT: highlight) the recipe is when it's available from the ready-panel, and I was of the opinion that that's the only time you would want to see it. I'm open to discussion on that one.

**Reply by OmikronGamma** [reply]
> Do you mean the icons that used to be at the top? I removed them because they seemed redundant with the new map suggestions panel. I assumed just following that and running optimal maps was better than checking the icons individually. The ready panel now tells you how close you are to completion of the prio-list (X bases missing). If you only want to loot missing bases, you can shift-click the MISSING-label to open a movable list that acts as a cheat sheet.

Now instead of glance-value, i have to shift click and deal with pop-up window. Thats bad change in my book. Would somebody really switch their maps "settings" (favourite maps and scarabs for them and whatnot) just to farm some ingredients?

> Hold-clicking shows the tree, normal clicking highlights the assembled version if one is available in the inventory.

I see. But why hold?

Nah, back to 1.21.0

PS. Oh, and i feel like new prio list (next to archnemesis inventory) is overburdened now, with pause, ready, not ready, burn lists. Too much.

**Reply by Lailloken** [reply]
> Now instead of glance-value, i have to shift click and deal with pop-up window. Thats bad change in my book.

Technically, that list has more glance-value since you can keep it open anywhere on the screen (it doesn't close automatically) and don't have to hover mods on the ground to compare their icons. Additionally, you had to open the inventory before to even see the icon list, now you just compare the names instead of icons. I'll think about reverting it, but the first reaction when I introduced the icons was people actually asking for names instead of icons.

> Would somebody really switch their maps "settings" (favourite maps and scarabs for them and whatnot) just to farm some ingredients

No. But what does that have to do with the icons being removed? I meant this
![Untitled 1](https://user-images.githubusercontent.com/61888437/156445000-b653e8c5-8f59-4f6e-a2b4-b80b2217c8c8.jpg).
Because if you don't use this, your best bet is RNG -- and the icon list doesn't really help with that. 

> I see. But why hold?

Because I never felt it had to stay open. You don't interact with it, just look at it. Otherwise it would be a click to open and a click to close it.

> PS. Oh, and i feel like new prio list (next to archnemesis inventory) is overburdened now, with pause, ready, not ready, burn lists

Thanks for your feedback. I'll see if I can slim it down a little. It's just that people keep asking for more features once you open the door by just a tiny crack, and I have to make it work with limited possibilities. I actually didn't feel the latest version added a lot to it (just pause and where the available recipes belong to). It's a shame the changes made you revert to the old version immediately.

**Comment by epicmeatwad** [comment]
The releases keep getting better and better, thank you!

**Comment by SirGuySW** [comment]
I've noticed some functional differences between 1.21.3 and 1.20.8 that don't seem quite right. Screenshots: https://imgur.com/a/ZXk7Bcq

1.21.3 lists some mods used in prio recipes as burnable (changing burn surplus value does not appear to change this (rescanned)). This can be observed in the last screenshot (3: 1.21.3) where Overcharged and Malediction are listed at the top of the burn list while they are components for mods I only have 1 of each. The second to last screenshot (3: 1.20.8) shows 1.20.8 not listing any burn mods (since all present mods are used in prio recipes and none are above the burn surplus threshold).

1.21.3 is also not recommending some ready recipes needed for the prio list. The second to last screenshot (3: 1.20.8) shows 1.20.8 recommends Magma Barrier, Ice Prison, and Corpse Detonator but the last screenshot (3: 1.21.3) shows 1.21.3 only recommends Ice Prison.

These are 'fresh installs' (extract to new folder, retrain, re-setup prio list).


Side note (probably intended but seems strange when considering the above): neither version recommends Drought Bringer (ready and I only have one) or recommends Bonebreaker as a burn mod despite having 4 (in the first comparison). This seems to indicate that it's reserving 4 Bonebreakers but none of Malediction or Overcharged.

Thanks for creating and sharing this UI. It made managing the league interface tolerable, and the league mechanic enjoyable.

**Reply by Lailloken** [reply]
Stellar reporting, very much appreciated.

> 1.21.3 lists some mods used in prio recipes as burnable

I tried to fix an oversight but mixed something up, my bad. I split the priority system internally into a complete priority list (which is checked before something is sent to the burn lists) and a copy that only contains priority mods that are explicitly missing from the chain to completion, mistakenly synchronizing them at some point.

> 1.21.3 is also not recommending some ready recipes needed for the prio list

Could you check if they are still needed? Checking the screenshot, I think I can make them out in your inventory already. That's what I fixed for 1.21.3 (it was actually intended to work like this since 1.15), i.e. not assembling sub-components unless they are missing in the branch. This is to reduce producing surplus in order to speed up undersupplied branches in situations like these (simplified example using bases A, B, C; can be applied to assembled mods A, B, C as well):
- Two sub-components (recipes A+B and B+C, respectively) are in some form part of the priority list
- B+C is ready, you assemble it; A+B is missing A which doesn't drop for a prolongued time
- another B+C is ready, still suggested, you assemble it again (even though it's not strictly necessary anymore); now B reserve is depleted
- you finally drop A, but now B stops dropping for a while, which creates unnecessary downtime for the prio-list as a whole that could have been avoided by not overproducing B+C

This prioritizes completing the whole prio-list first over pumping out products that are part of it. Please let me know if that's a good idea or if I should revert this behavior.

> This seems to indicate that it's reserving 4 Bonebreakers but none of Malediction or Overcharged.

As far as I can tell, this is a byproduct of the mistake mentioned in the first paragraph. In this case, Bonebreaker is part of tukohama-touched which is not assembled yet. This means bonebreaker stays in the priority list, whereas malediction and overcharge are mistakenly omitted because the product they are a part of is already assembled.

Thanks again for your thoroughness. I just woke up and was immediately able to see what's wrong thanks to your feedback.

**Reply by SirGuySW** [reply]
> Could you check if they are still needed? Checking the screenshot, I think I can make them out in your inventory already. That's what I fixed for 1.21.3

That seems to be working as intended then (and thank you for the extra explanation). To reinforce the explanation I used my last Drought Bringer and it is now being recommended again. I think this style is probably best for most uses/users. Unfortunately I have a bunch of prioritized recipes that I only want to complete once (for the challenges), while others I want to complete repeatedly. I wouldn't mind building a surplus of sub-components for those main recipes while I wait on the Frenzieds needed for the 'only once' recipes. More importantly: I'd rather not burn components for my main recipes while I have duplicate components for the 'only once' recipes (which I'll never actually use to build into the recipe components and can be safely burned).

I think what I'll do moving forward is just omit (or pause, I haven't used that yet, looks promising) those 'only once' recipes, make a note of the components that are holding them back (Frenzied), and make a note of already assembled components to ignore in the mean time ("don't burn these under any circumstances"). That should allow me to eventually build Innocence/Kitava without them bogging down the other recipes.

> Thanks again for your thoroughness. I just woke up and was immediately able to see what's wrong thanks to your feedback.

I'm very glad you found this constructive. Thank you for letting me know. :)
